org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/core/manipulation/CleanUpContextCore.java

```diff
@@ -24,7 +24,9 @@
  * The context that contains all information required by a clean up to create a fix.
  *
  * @since 1.12
+ * @deprecated Please use CleanUpContext in the future
  */
+@Deprecated
```

Comment:
As there is a clean/direct replacement please mark it as forRemoval=true, since="2024-03" so after 2026-03 release it can be deleted. Please apply elsewhere if you see fit.